From 5e6201b30380ed31604f4063bdd385c1907883df Mon Sep 17 00:00:00 2001
From: Lei Maohui <leimaohui@fujitsu.com>
Date: Wed, 15 Oct 2025 06:39:05 +0000
Subject: [PATCH] meson: Correct installation umask to 0022.

The default installation umask of 0027 for Kea-built artifacts, particularly for directories like /usr/lib, results in restrictive permissions (0750, drwxr-x---).This deviates from the common Linux and Yocto standard of 0755 (drwxr-xr-x) for shared library directories.

Signed-off-by: Lei Maohui <leimaohui@fujitsu.com>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index f1016e2..af7b864 100644
--- a/meson.build
+++ b/meson.build
@@ -26,7 +26,7 @@ project(
         'b_lundef=false',  # required for clang++ when used with sanitizers and for BSD ld when linking with extern char **environ.
         'b_pch=false',
         'b_pie=true',
-        'install_umask=0027',
+        'install_umask=0022',
         'warning_level=2',
     ],
 )
-- 
2.43.0

